docs/developer/architecture/core/packages.asciidoc

```diff
@@ -0,0 +1,26 @@
+[[core-packages]]
+== Core packages
+
+experimental[]
+
+Core packages have well defined boundaries, have a single responsibility, and are organized by domain. Core packages follow a specific naming schema, according to what the contain:
+
+For example, core capapability packages are:
+
+* `core-capabilities-browser-internal`
+* `core-capabilities-browser-mocks`
+* `core-capabilities-common`
+* `core-capabilities-server`
+* `core-capabilities-server-internal`
+* `core-capabilities-server-mocks`
+
+Every domain has a specific package for public types, importable from everywhere in Kibana code, implementation and unit tests, internal to core, not intended for public use, and mocks, exposed to plugins for unit testing.
```

Comment:
```suggestion
Each domain has a specific package for public types, which can be imported and used throughout the Kibana codebase including in its implementation and unit tests. These packages are internal to core and not intended for public use, but they can be used by plugins to create mock versions for unit testing.
```

Reply:
If I've misunderstood this please feel free to ignore the suggestion.

Reply:
You understand perfectly 😄